### ISSUE TITLE
Always scroll to source directory when walking up in list

### ISSUE THREAD

**Issue by mowgli** [BestImageViewer/geeqie#351]
When walking ones directory tree in list mode selecting "..", sometimes the views scrolls to the directory from where you came and sometimes it does not.
Attached is a small patch that makes sure the view always scrolls to the directory from where we came.
Apparently the problem was that the source directory entry name sometimes is "..", not very informative when searching the new directory contents after an entry with that name, so changed to search after path instead.

Reported by: the_jk

**Comment by mowgli** [comment]
Original patch:
```patch
Index: src/view_dir_list.c
===================================================================
--- src/view_dir_list.c	(revision 1917)
+++ src/view_dir_list.c	(working copy)
@@ -309,7 +309,7 @@
 		base = remove_level_from_path(vd->dir_fd->path);
 		if (strcmp(base, dir_fd->path) == 0)
 			{
-			old_path = g_strdup(vd->dir_fd->name);
+			old_path = g_strdup(vd->dir_fd->path);
 			}
 		g_free(base);
 		}
@@ -329,7 +329,7 @@
 		while (work && !found)
 			{
 			FileData *fd = work->data;
-			if (strcmp(old_path, fd->name) == 0) found = fd;
+			if (strcmp(old_path, fd->path) == 0) found = fd;
 			work = work->next;
 			}
 
```

**Comment by mowgli** [comment]
Should be fixed in 8cd21d2.

But will have an eye for the second part...
